File: Lab04 JAVA/Desenvolvedor.java
public class Desenvolvedor extends Funcionario implements InterfaceDesenvolvedor {
    private String linguagemPrincipal;

    public Desenvolvedor(String nome, double salarioBase, String linguagemPrincipal) {
        super(nome, salarioBase);
        this.linguagemPrincipal = linguagemPrincipal;
    }

    public String getLinguagemPrincipal() {
        return linguagemPrincipal;
    }

    public void setLinguagemPrincipal(String linguagemPrincipal) {
        this.linguagemPrincipal = linguagemPrincipal;
    }

    public String projetosRealizados() {
        return "Projetos entregues na linguagem " + linguagemPrincipal;
    }
    
    @Override
    public String descricao() {
    return super.descricao() + ", Linguagem principal: " + linguagemPrincipal;
}
}

File: Labs/Lab04 C++/Desenvolvedor.cpp
#include "Desenvolvedor.hpp"

using namespace std;

Desenvolvedor::Desenvolvedor(const string &nome, int salario_base, const std::string &linguagem_principal)
    : Funcionario(nome, salario_base), linguagem_principal(linguagem_principal) {}

string Desenvolvedor::getLinguagemPrincipal() const
{
    return linguagem_principal;
}

string Desenvolvedor::projetos_realizados() const
{
    return "Projetos realizados na linguagem:" + linguagem_principal;
}